core/src/main/java/hudson/tools/ToolInstallation.java

```diff
@@ -82,8 +81,7 @@
     /**
      * {@link ToolProperty}s that are associated with this tool.
      */
-    @XStreamSerializable
-    private transient /*almost final*/ DescribableList<ToolProperty<?>,ToolPropertyDescriptor> properties
+    private /*almost final*/ DescribableList<ToolProperty<?>,ToolPropertyDescriptor> properties
```

Comment:
This is a problem. `DescribableList` is not `Serializable` so removing `transient` will prevent this class from being serialized over Remoting. Yet keeping `transient` will prevent tool installations from saving their properties in XML. I have not found any way to get the effect of `XStreamSerializable` without patching `PureJavaReflectionProvider`.

Reply:
(I am not sure why `ToolInstallation` was ever made `Serializable` to begin with. The normal usage is to have it define some `PATH` entry on the controller side, which does not require the `ToolInstallation` itself to ever be sent to the agent.)

Reply:
Also the current annotation was explicitly rejected upstream: https://github.com/x-stream/xstream/pull/70#issuecomment-254899878